Remake Puppet introduction (#1039)
theforeman/foreman-documentation: guides/common/modules/ref_performing-configuration-management.adoc

[id="performing-configuration-management_{context}"]
= Performing Configuration Management

Performing configuration management using Puppet in {Project} involves the following high-level steps:

. Managing Puppet modules on the Puppet server, i.e. installing and updating them.
. Importing Puppet classes and environments from Puppet modules into {Project}.
. Optional: Creating config groups.
. Configuring overrides of Smart Class parameters on various levels.
. Assigning Puppet classes, modules or config groups to host groups or individual hosts.
. Configuring intervals for runs of the Puppet agent on hosts and for configuration enforcement runs of the Puppet server.
. Monitoring configuration management via reports in the {ProjectWebUI}.
. Optional but recommended: Configuring email notifications.
See {AdministeringDocURL}Configuring_Email_Notifications_admin[Configuring Email Notifications] in _Administering {ProjectName}_.

> {Project} then runs configuration management automatically in the configured intervals to enforce Puppet configuration on the managed hosts, or you can initiate it manually on demand with the *Run Puppet Once* feature.

Reviewer: The automatical run happens only once user deploy the Puppet agent on the host, that typically happens as part of the provisioning process. Otherwise, users must install the agent manually (or automate in the registration template, which should be rather trivial)

Author: That's the assumption of this module. I've tried to add a mention into the first paragraph.

[id="performing-configuration-management_{context}"]
= Performing Configuration Management

Once you deployed Puppet agent on a host, you can start performing configuration management with Puppet.
This involves the following high-level steps:

> . Managing Puppet modules on the Puppet server, i.e. installing and updating them.

Reviewer 2: ```suggestion
. Managing Puppet modules on the Puppet server, that means installing and updating them.
```
I think `i.e.` is forbidden.

. Importing Puppet classes and environments from Puppet modules into {Project}.
. Optional: Creating config groups.
. Configuring overrides of Smart Class parameters on various levels.

> . Assigning Puppet classes, modules or config groups to host groups or individual hosts.

Reviewer 2: ```suggestion
. Assigning Puppet classes, modules, or config groups to host groups or individual hosts.
```

. Configuring intervals for runs of the Puppet agent on hosts and for configuration enforcement runs of the Puppet server.

> . Monitoring configuration management via reports in the {ProjectWebUI}.

Reviewer 2: Please reword "via" because it's latin.

> . Optional but recommended: Configuring email notifications.

Reviewer 3: Ideally we just say do it if we want them to do it regardless of it is optional.
